crates/daphne/src/roles/helper/handle_agg_job.rs

```diff
@@ -0,0 +1,321 @@
+    task_config: DapTaskConfig,
+    reports: Vec<InitializedReport<WithPeerPrepShare>>,
+}
+
+macro_rules! impl_from {
+    ($($t:ty),*$(,)?) => {
+        $(impl From<$t> for HandleAggJob<$t> {
+            fn from(state: $t) -> Self {
+                Self { state }
+            }
+        })*
+    };
+}
+
+impl_from!(Init, WithTaskConfig, InitializedReports);
+
+pub fn start(request: DapRequest<AggregationJobInitReq>) -> HandleAggJob<Init> {
+    HandleAggJob::new(request)
+}
+
+impl HandleAggJob<Init> {
+    pub fn new(request: DapRequest<AggregationJobInitReq>) -> Self {
+        Self {
+            state: Init(request),
+        }
+    }
+
+    /// Resolve the task config in the default way.
+    pub async fn resolve_task_config<A: DapHelper>(
+        self,
+        aggregator: &A,
+    ) -> Result<HandleAggJob<WithTaskConfig>, DapError> {
+        let task_config = resolve_task_config(aggregator, &self.state.0.meta).await?;
+        self.with_task_config(task_config)
+    }
+
+    /// Provide the resolved task configuration.
+    pub fn with_task_config(
+        self,
+        task_config: DapTaskConfig,
+    ) -> Result<HandleAggJob<WithTaskConfig>, DapError> {
+        let Self {
+            state: Init(request),
+        } = self;
+
+        check_part_batch(
+            &request.task_id,
+            &task_config,
+            &request.payload.part_batch_sel,
+            &request.payload.agg_param,
+        )?;
+        Ok(HandleAggJob {
+            state: WithTaskConfig {
+                task_config,
+                request,
+            },
+        })
+    }
+}
+
+impl HandleAggJob<WithTaskConfig> {
+    /// Initialize the reports in the default way.
+    pub async fn initialize_reports<A: DapHelper>(
+        self,
+        aggregator: &A,
+        replay_protection: crate::ReplayProtection,
+    ) -> Result<HandleAggJob<InitializedReports>, DapError> {
+        let WithTaskConfig {
+            task_config,
+            request,
+        } = self.state;
+        let task_id = request.task_id;
+        let part_batch_sel = request.payload.part_batch_sel.clone();
+        let initialized_reports = task_config.consume_agg_job_req(
+            &aggregator.get_receiver_configs(task_config.version).await?,
+            aggregator.valid_report_time_range(),
+            &task_id,
+            request.payload,
+            replay_protection,
+        )?;
+
+        Ok(HandleAggJob {
+            state: InitializedReports {
+                task_id,
+                task_config,
+                part_batch_sel,
+                reports: initialized_reports,
+            },
+        })
+    }
+
+    /// Splits the state machine into it's component parts, giving the caller more control over how
+    /// the initialization of reports should be done.
+    ///
+    /// To return to the [`HandleAggJob`] state machine flow, call
+    /// [`ToInitializedReportsTransition::with_initialized_reports`].
+    pub fn into_parts(
```

Comment:
"parts" is a little vague
```suggestion
    pub fn into_initialized_reports_and_request(
```

Reply:
`into_initialized_reports` is misleading, this doesn't return initialized_reports, it returns the types needed for report initialization. I preferred to go with a shorter "into_parts" like the [the std lib does](https://doc.rust-lang.org/std/index.html?search=_parts) since use of this method requires reading the documentation anyway.